src/components/GameItem.js

```diff
@@ -4,23 +4,43 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
-  propTypes = {
+  static propTypes = {
     height: PropTypes.number.isRequired,
     layer: PropTypes.number.isRequired,
   }
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      clicked: false
+    };
+  }
+
+  clickedItem = () => {
+    this.setState({
+      clicked: true,
+    });
+    this.props.type === 'litter' && this.props.clickedItemClickCallback(this.props.index);
```

Comment:
Hmmm, what does this code do?
This code reads as "If the following is true: ... `this.props.type` is equal to "litter" and also invoke this function named `this.props.clickedItemClickCallback()`, then... do nothing!". You're doing a boolean expression. It just so happens that the code is executing it, going, "hm, if `this.props.type === 'litter'`, and let me check if the other side is true" and invoking the function.

It likely will make more sense if the code looks like this:
```javascript
if (this.props.type === 'litter') {
  this.props.clickedItemClickCallback(this.props.index);
}
```
so then it'll read "if the first part is true, then do the stuff inside."